base/math.jl

```diff
@@ -293,10 +293,10 @@ end
 
 # polynomial evaluation using compensated summation.
 # much more accurate, especially when lo can be combined with other rounding errors
-@inline function exthorner(x, p::Tuple)
+Base.@assume_effects :terminates_locally  @inline function exthorner(x::Union{Float16, Float32, Float64}, p::Tuple)
```

Comment:
Why the extra type restriction?

Reply:
when I marked it `:total` I wanted to be slightly more cautions, but now that it's only `:terminates_locally` I'll relax it again.